pkg/sentry/event/beacon/eth/v2/validator_beacon_block.go

```diff
@@ -0,0 +1,203 @@
+package event
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/attestantio/go-eth2-client/api"
+	"github.com/attestantio/go-eth2-client/spec"
+	"github.com/ethpandaops/xatu/pkg/proto/eth"
+	"github.com/ethpandaops/xatu/pkg/proto/xatu"
+	"github.com/ethpandaops/xatu/pkg/sentry/ethereum"
+	ssz "github.com/ferranbt/fastssz"
+	"github.com/golang/snappy"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+type ValidatorBeaconBlock struct {
+	log        logrus.FieldLogger
+	snapshot   *ValidatorBeaconBlockDataSnapshot
+	event      *api.VersionedProposal
+	beacon     *ethereum.BeaconNode
+	clientMeta *xatu.ClientMeta
+	id         uuid.UUID
+}
+
+type ValidatorBeaconBlockDataSnapshot struct {
+	RequestAt       time.Time
+	RequestDuration time.Duration
+}
+
+func NewValidatorBeaconBlock(log logrus.FieldLogger, event *api.VersionedProposal, snapshot *ValidatorBeaconBlockDataSnapshot, beacon *ethereum.BeaconNode, clientMeta *xatu.ClientMeta) *ValidatorBeaconBlock {
+	return &ValidatorBeaconBlock{
+		log:        log.WithField("event", "BEACON_API_ETH_V3_VALIDATOR_BEACON_BLOCK"),
+		event:      event,
+		snapshot:   snapshot,
+		beacon:     beacon,
+		clientMeta: clientMeta,
+		id:         uuid.New(),
+	}
+}
+
+func (e *ValidatorBeaconBlock) Decorate(ctx context.Context) (*xatu.DecoratedEvent, error) {
+	data, err := eth.NewEventBlockV2FromVersionSignedBeaconBlock(e.event)
```

Comment:
Happy for me to refactor to handle `VersionedSignedBeaconBlock`'s and `VersionedProposal`'s?

Reply:
Discussed, refactor.